pkg/controllers/repositoryserver/handler.go

```diff
@@ -171,20 +176,26 @@ func (h *RepoServerHandler) reconcilePod(ctx context.Context, svc *corev1.Servic
 		return nil, nil, err
 	}
 	h.Logger.Info("Pod resource not found. Creating new pod")
+	return h.createPodAndUpdateStatus(ctx, repoServerNamespace, svc)
+}
+
+func (h *RepoServerHandler) setCondition(ctx context.Context, conditionStatus ConditionStatus) error {
```

Comment:
```suggestion
func (h *RepoServerHandler) setCondition(ctx context.Context, condition metav1.Condition, progress WhatEverTypeThisIs) error {
```
This is going to simplify a lot of things. We wont have to form the arguments first and then figure out metav1.Condition from those arguments. We can just get the metav1.Condition and pass that around.

Reply:
we are forming metav1.Condition from the getCondition() functions inside, If we want to pass metav1.condition, we would not maybe need the function also.

Reply:
this function was added to replace duplicate calls for getCondition() and updatestatus() in the Reconciliation.

Reply:
Right, so we should just remove that function? And that's the reason I suggested this change, to make things simpler and don't have extra type.

Reply:
Ok I can do this but then we would have duplicate calls in Reconciliation and the function would be long, I had added this to address other comments that was to move duplicate things to a function. Would that be ok?

Reply:
just want to make sure we are on the same page.

Reply:
duplicated calls to which function?

Reply:
I have addressed those comments please check.